samples/tutorials/Data passing in python components/Data passing in python components - Files.py

```diff
@@ -16,149 +16,130 @@
-def sum_pipeline(count: int = 100000):
-    numbers_task = write_numbers(count=count)
-    print_text(numbers_task.output)
-
-    sum_task = sum_numbers(numbers_task.outputs['numbers'])
-    print_text(sum_task.output)
-
-
-# Submit the pipeline for execution:
-#kfp.Client(host=kfp_endpoint).create_run_from_pipeline_func(sum_pipeline, arguments={})
 
-# Combining all pipelines together in a single pipeline
-def file_passing_pipelines():
-    print_repeating_lines_pipeline()
-    text_splitting_pipeline()
-    sum_pipeline()
+from typing import Dict, List
+
+from kfp import compiler
+from kfp import dsl
+from kfp.dsl import Input, InputPath, Output, OutputPath, Dataset, Model, component
+
+
+@component
+def preprocess(
+    # An input parameter of type string.
+    message: str,
+    # Use Output[T] to get a metadata-rich handle to the output artifact
+    # of type `Dataset`.
+    output_dataset_one: Output[Dataset],
+    # A locally accessible filepath for another output artifact of type
+    # `Dataset`.
+    output_dataset_two_path: OutputPath('Dataset'),
+    # A locally accessible filepath for an output parameter of type string.
+    output_parameter_path: OutputPath(str),
+    # A locally accessible filepath for an output parameter of type bool.
+    output_bool_parameter_path: OutputPath(bool),
+    # A locally accessible filepath for an output parameter of type dict.
+    output_dict_parameter_path: OutputPath(Dict[str, int]),
+    # A locally accessible filepath for an output parameter of type list.
+    output_list_parameter_path: OutputPath(List[str]),
+):
+    """Dummy preprocessing step."""
+
+    # Use Dataset.path to access a local file path for writing.
+    # One can also use Dataset.uri to access the actual URI file path.
+    with open(output_dataset_one.path, 'w') as f:
+        f.write(message)
+
+    # OutputPath is used to just pass the local file path of the output artifact
+    # to the function.
+    with open(output_dataset_two_path, 'w') as f:
+        f.write(message)
+
+    with open(output_parameter_path, 'w') as f:
+        f.write(message)
+
+    with open(output_bool_parameter_path, 'w') as f:
+        f.write(
+            str(True))  # use either `str()` or `json.dumps()` for bool values.
+
+    import json
+    with open(output_dict_parameter_path, 'w') as f:
+        f.write(json.dumps({'A': 1, 'B': 2}))
+
+    with open(output_list_parameter_path, 'w') as f:
+        f.write(json.dumps(['a', 'b', 'c']))
+
+
+@component
+def train(
+    # Use InputPath to get a locally accessible path for the input artifact
+    # of type `Dataset`.
+    dataset_one_path: InputPath('Dataset'),
+    # Use Input[T] to get a metadata-rich handle to the input artifact
+    # of type `Dataset`.
+    dataset_two: Input[Dataset],
+    # An input parameter of type string.
+    message: str,
+    # Use Output[T] to get a metadata-rich handle to the output artifact
+    # of type `Dataset`.
+    model: Output[Model],
```

Comment:
Should this be `model: Output[Dataset]`?

Reply:
No, it's intended to be `model: Output[Model]`. What makes you think it should be `Dataset`?

Reply:
Fixed the comment. Thanks for catching this.